Update DEPLOYMENT.md

henrikstranneheim opened the pull request:
This PR adds/fixes:

- Update deployments instructions
- Lock some python packages

**How to prepare for test**:
- [ ] ssh to Hasta
- [ ] install on stage of Hasta:
`bash servers/resources/hasta.sclifelab.se/update-genotype-stage.sh Use-new-generic-update-script`

**How to test**:
- [ ] login to ...
- [ ] click on ...

**Expected test outcome**:
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by @ivadym 
- [x] tests executed by HS
- [x] "Merge and deploy" approved by HS
Thanks for filling in who performed the code review and the test!

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

Implementation plan:
- [ ] Deploy to ...
- [ ] Inform to ...

henrikstranneheim commented:
Tested here: https://github.com/Clinical-Genomics/servers/pull/910

ivadym reviewed (approved):
Nothing to add from my part 💯

henrikstranneheim commented on .github/PULL_REQUEST_TEMPLATE:
```diff
@@ -3,19 +3,18 @@ This PR adds/fixes ...
 - [ ] ssh to Hasta
 - [ ] install on stage of Hasta:
-`bash servers/resources/hasta.sclifelab.se/update-genotype-stage.sh [THIS-BRANCH-NAME]`
+`bash servers/resources/hasta.sclifelab.se/update-tool-stage.sh -e S_genotype -t genotype -b [THIS-BRANCH-NAME]`
```
```suggestion
`bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_genotype -t genotype -b [THIS-BRANCH-NAME]`
```

henrikstranneheim commented:
``` Install in stage using new binary ✅ 
θ61° [henrik.stranneheim@hasta:~] [base] 5s $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_genotype -t genotype -b Use-new-generic-update-script
Collecting git+https://github.com/Clinical-Genomics/genotype@Use-new-generic-update-script
  Running command git clone --quiet https://github.com/Clinical-Genomics/genotype /home/henrik.stranneheim/tmp/pip-req-build-63l8l2sr
  Cloning https://github.com/Clinical-Genomics/genotype (to revision Use-new-generic-update-script) to ./tmp/pip-req-build-63l8l2sr
  Running command git checkout -b Use-new-generic-update-script --track origin/Use-new-generic-update-script
  Resolved https://github.com/Clinical-Genomics/genotype to commit 2c5ba4864d540433e71503ae549b6efaf3a01f61
  Switched to a new branch 'Use-new-generic-update-script'
  Installing build dependencies: started
  Branch Use-new-generic-update-script set up to track remote branch Use-new-generic-update-script from origin.
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Building wheels for collected packages: genotype
  Building wheel for genotype (pyproject.toml): started
  Building wheel for genotype (pyproject.toml): finished with status 'done'
  Created wheel for genotype: filename=genotype-4.1.0-py3-none-any.whl size=31299 sha256=1767b019eba6cebfdad0465c262a8576b168fd08c853f2706882762ea40146a2
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-0mz_j0f4/wheels/b4/bd/1f/de9d4c5659600d3be37b9eb80c14019c93b33ac93df02bdc68
Successfully built genotype
Installing collected packages: genotype
  Attempting uninstall: genotype
    Found existing installation: genotype 4.1.0
    Uninstalling genotype-4.1.0:
      Successfully uninstalled genotype-4.1.0
Successfully installed genotype-4.1.0
Collecting git+https://github.com/Clinical-Genomics/genotype@Use-new-generic-update-script
  Running command git clone --quiet https://github.com/Clinical-Genomics/genotype /home/henrik.stranneheim/tmp/pip-req-build-p0j0gc62
  Cloning https://github.com/Clinical-Genomics/genotype (to revision Use-new-generic-update-script) to ./tmp/pip-req-build-p0j0gc62
  Running command git checkout -b Use-new-generic-update-script --track origin/Use-new-generic-update-script
  Resolved https://github.com/Clinical-Genomics/genotype to commit 2c5ba4864d540433e71503ae549b6efaf3a01f61
  Switched to a new branch 'Use-new-generic-update-script'
  Installing build dependencies: started
  Branch Use-new-generic-update-script set up to track remote branch Use-new-generic-update-script from origin.
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Requirement already satisfied: pyyaml in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (6.0)
Requirement already satisfied: wheel in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.37.1)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (2.1)
Requirement already satisfied: xlrd==1.2.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (1.2.0)
Requirement already satisfied: pymysql in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (1.0.2)
Requirement already satisfied: ruamel.yaml in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.17.21)
Requirement already satisfied: Flask-Reverse-Proxy in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.2.0.2)
Requirement already satisfied: Flask-Bootstrap in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (3.3.7.1)
Requirement already satisfied: gunicorn in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (20.1.0)
Requirement already satisfied: Flask-Alchy in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.5.0)
Requirement already satisfied: cyvcf2 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.30.18)
Requirement already satisfied: Flask-Login==0.5.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.5.0)
Requirement already satisfied: click in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (8.1.3)
Collecting numpy==1.19.2
  Using cached numpy-1.19.2-cp37-cp37m-manylinux2010_x86_64.whl (14.5 MB)
Requirement already satisfied: Flask-Dance==3.0.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (3.0.0)
Requirement already satisfied: markupsafe==1.1.1 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (1.1.1)
Requirement already satisfied: jinja2==2.11.2 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (2.11.2)
Requirement already satisfied: Alchy in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (2.2.2)
Requirement already satisfied: coloredlogs in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (15.0.1)
Requirement already satisfied: itsdangerous==1.1.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (1.1.0)
Requirement already satisfied: cython in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.29.32)
Requirement already satisfied: Flask-Security in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (3.0.0)
Requirement already satisfied: SQLAlchemy<1.4 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (1.3.24)
Requirement already satisfied: werkzeug<1.0.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.16.1)
Requirement already satisfied: Flask==1.1.1 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (1.1.1)
Requirement already satisfied: pysam in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from genotype==4.1.0) (0.19.1)
Requirement already satisfied: urlobject in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.0) (2.4.3)
Requirement already satisfied: six in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.0) (1.16.0)
Requirement already satisfied: requests-oauthlib>=1.0.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.0) (1.3.1)
Requirement already satisfied: oauthlib in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.0) (3.2.0)
Requirement already satisfied: requests>=2.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.0) (2.28.1)
Requirement already satisfied: importlib-metadata in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from click->genotype==4.1.0) (5.0.0)
Requirement already satisfied: pydash>=3.4.7 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Alchy->genotype==4.1.0) (5.1.0)
Requirement already satisfied: humanfriendly>=9.1 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from coloredlogs->genotype==4.1.0) (10.0)
Requirement already satisfied: dominate in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Bootstrap->genotype==4.1.0) (2.7.0)
Requirement already satisfied: visitor in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Bootstrap->genotype==4.1.0) (0.1.3)
Requirement already satisfied: passlib>=1.7 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.0) (1.7.4)
Requirement already satisfied: Flask-BabelEx>=0.9.3 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.0) (0.9.4)
Requirement already satisfied: Flask-WTF>=0.13.1 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.0) (1.0.1)
Requirement already satisfied: Flask-Principal>=0.3.3 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.0) (0.4.0)
Requirement already satisfied: Flask-Mail>=0.7.3 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.0) (0.9.1)
Requirement already satisfied: setuptools>=3.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from gunicorn->genotype==4.1.0) (65.3.0)
Requirement already satisfied: ruamel.yaml.clib>=0.2.6 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from ruamel.yaml->genotype==4.1.0) (0.2.6)
Requirement already satisfied: Babel>=1.0 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.0) (2.10.3)
Requirement already satisfied: speaklater>=1.2 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.0) (1.3)
Requirement already satisfied: blinker in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-Mail>=0.7.3->Flask-Security->genotype==4.1.0) (1.5)
Requirement already satisfied: WTForms in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Flask-WTF>=0.13.1->Flask-Security->genotype==4.1.0) (3.0.1)
Requirement already satisfied: urllib3<1.27,>=1.21.1 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.0) (1.26.12)
Requirement already satisfied: charset-normalizer<3,>=2 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.0) (2.1.1)
Requirement already satisfied: certifi>=2017.4.17 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.0) (2022.6.15)
Requirement already satisfied: idna<4,>=2.5 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.0) (3.3)
Requirement already satisfied: zipp>=0.5 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from importlib-metadata->click->genotype==4.1.0) (3.10.0)
Requirement already satisfied: typing-extensions>=3.6.4 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from importlib-metadata->click->genotype==4.1.0) (4.4.0)
Requirement already satisfied: pytz>=2015.7 in /home/proj/stage/bin/miniconda3/envs/S_genotype/lib/python3.7/site-packages (from Babel>=1.0->Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.0) (2022.2.1)
Installing collected packages: numpy
  Attempting uninstall: numpy
    Found existing installation: numpy 1.21.6
    Uninstalling numpy-1.21.6:
      Successfully uninstalled numpy-1.21.6
Successfully installed numpy-1.19.2
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/henrik.stranneheim
done.
Log deploy... done.
θ64° [henrik.stranneheim@hasta:~] [base] 1m34s $ which genotype
alias genotype='/home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml'
	/home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype
θ64° [henrik.stranneheim@hasta:~] [base] $ gentype --version
bash: gentype: command not found...
θ60° [henrik.stranneheim@hasta:~] [base] 127 $ genotype --version
genotype, version 4.1.0
θ63° [henrik.stranneheim@hasta:~] [base] 5s $ genotype --help
Usage: genotype [OPTIONS] COMMAND [ARGS]...

  Interact with genotype comparison tool.

Options:
  -c, --config PATH     path to config file
  -d, --database TEXT   path/URI of the SQL database
  -l, --log-level TEXT
  --log-file PATH
  --version             Show the version and exit.
  --help                Show this message and exit.

Commands:
  add-sex                 Add sex determination to samples and analyses.
  check                   Check integrity of a sample.
  delete                  Delete analyses and samples from the database.
  delete-user             Delete user from the database.
  export-sample           Gets data for samples from the sample table,...
  export-sample-analysis  Gets analysis data for samples from the...
  init                    Setup a new Genotype database.
  list-users              List users from the database.
  load                    Load data from genotype resources.
  ls                      List samples from the database.
  match                   Match genotypes for an analysis against all...
  sample                  Get a sample from the database.
  serve                   Start up the web interface.
  snps                    Export the snp definitions
  view                    View added genotypes for all analyses of a sample.
```

## Upload ✅ 

```
θ65° [henrik.stranneheim@hasta:~] [base] 24s $ cg upload genotypes justhusky
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:2231: SAWarning: Cascade settings "delete, merge, save-update" should not be combined with a viewonly=True relationship.   This configuration will raise an error in version 1.4.  Note that in versions prior to 1.4, these cascade settings may still produce a mutating effect even though this relationship is marked as viewonly=True.
  "viewonly=True." % (", ".join(sorted(non_viewonly)))
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
Called undefined __fields__ on HousekeeperAPI, please wrap
----------------- UPLOAD -----------------
----------------- GENOTYPES -------------------
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Demux.samples to refer to viewonly relationship Sample.demuxes should include sync_backref=False set on the Demux.samples relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
/home/proj/stage/bin/miniconda3/envs/S_main/lib/python3.7/site-packages/sqlalchemy/orm/relationships.py:1998: SAWarning: Setting backref / back_populates on relationship Sample.demuxes to refer to viewonly relationship Demux.samples should include sync_backref=False set on the Sample.demuxes relationship.  (this warning may be suppressed after 10 occurrences)
  (rel_b, rel_a, rel_b),
Initializing UploadGenotypesAPI
Fetching upload genotype data for justhusky
Fetch latest version from bundle justhusky
Fetching files with tags in [genotype]
Fetching files from version 111220
Fetching files with tags in [qc-metrics,deliverable]
Fetching files from version 111220
loading VCF genotypes for sample(s): earlycasualcaiman, slowlycivilbuck, hugelymodelbat
Running command /home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml load /home/proj/stage/housekeeper-bundles/justhusky/2022-09-22/justhusky_gatkcomb.bcf
Running command /home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml add-sex earlycasualcaiman -s male
Running command /home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml add-sex earlycasualcaiman -a sequence male
Running command /home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml add-sex slowlycivilbuck -s female
Running command /home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml add-sex slowlycivilbuck -a sequence female
Running command /home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml add-sex hugelymodelbat -s male
Running command /home/proj/stage/bin/miniconda3/envs/S_genotype/bin/genotype --config /home/proj/stage/servers/config/hasta.scilifelab.se/genotype-stage.yaml add-sex hugelymodelbat -a sequence male
```

henrikstranneheim commented:
Deployed on Hasta:

```
θ71° [henrik.stranneheim@hasta:~] [P_main] $ bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-genotype-prod.sh
on the host hasta.scilifelab.se, continue
Collecting git+https://github.com/clinical-genomics/genotype@master
  Cloning https://github.com/clinical-genomics/genotype (to revision master) to ./tmp/pip-req-build-k4_oyj6b
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Building wheels for collected packages: genotype
  Building wheel for genotype (PEP 517) ... done
  Created wheel for genotype: filename=genotype-4.1.1-py3-none-any.whl size=31298 sha256=6ec16b752159fb063a4b736926f32638972b35e9d9035028b29b32c286ca028f
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-dm2j4qzf/wheels/63/d7/38/2d616f160ea626015965a94d5e26e4288d169a24a3f5151da1
Successfully built genotype
Installing collected packages: genotype
  Attempting uninstall: genotype
    Found existing installation: genotype 4.1.0
    Uninstalling genotype-4.1.0:
      Successfully uninstalled genotype-4.1.0
Successfully installed genotype-4.1.1
Collecting git+https://github.com/clinical-genomics/genotype@master
  Cloning https://github.com/clinical-genomics/genotype (to revision master) to ./tmp/pip-req-build-qm_qku_u
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied, skipping upgrade: pyyaml in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (5.3.1)
Requirement already satisfied, skipping upgrade: ruamel.yaml in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.16.12)
Requirement already satisfied, skipping upgrade: xlrd==1.2.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (1.2.0)
Requirement already satisfied, skipping upgrade: Flask-Login==0.5.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.5.0)
Requirement already satisfied, skipping upgrade: gunicorn in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (20.1.0)
Requirement already satisfied, skipping upgrade: Flask-Security in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (3.0.0)
Requirement already satisfied, skipping upgrade: SQLAlchemy<1.4 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (1.3.19)
Requirement already satisfied, skipping upgrade: pymysql in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.10.1)
Requirement already satisfied, skipping upgrade: Flask-Alchy in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.5.0)
Requirement already satisfied, skipping upgrade: wheel in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.35.1)
Requirement already satisfied, skipping upgrade: Flask==1.1.1 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (1.1.1)
Requirement already satisfied, skipping upgrade: cython in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.29.21)
Requirement already satisfied, skipping upgrade: itsdangerous==1.1.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (1.1.0)
Requirement already satisfied, skipping upgrade: jinja2==2.11.2 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (2.11.2)
Requirement already satisfied, skipping upgrade: Alchy in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (2.2.2)
Requirement already satisfied, skipping upgrade: markupsafe==1.1.1 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (1.1.1)
Requirement already satisfied, skipping upgrade: click in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (7.1.2)
Requirement already satisfied, skipping upgrade: cyvcf2 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.9.0)
Requirement already satisfied, skipping upgrade: werkzeug<1.0.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.16.1)
Requirement already satisfied, skipping upgrade: Flask-Dance==3.0.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (3.0.0)
Requirement already satisfied, skipping upgrade: pysam in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.15.2)
Requirement already satisfied, skipping upgrade: coloredlogs in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (14.0)
Requirement already satisfied, skipping upgrade: Flask-Reverse-Proxy in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (0.2.0.2)
Requirement already satisfied, skipping upgrade: numpy==1.19.2 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (1.19.2)
Requirement already satisfied, skipping upgrade: Flask-Bootstrap in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (3.3.7.1)
Requirement already satisfied, skipping upgrade: Flask-SQLAlchemy==2.1 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from genotype==4.1.1) (2.1)
Requirement already satisfied, skipping upgrade: ruamel.yaml.clib>=0.1.2; platform_python_implementation == "CPython" and python_version < "3.9" in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from ruamel.yaml->genotype==4.1.1) (0.2.2)
Requirement already satisfied, skipping upgrade: setuptools>=3.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from gunicorn->genotype==4.1.1) (49.6.0.post20200814)
Requirement already satisfied, skipping upgrade: Flask-BabelEx>=0.9.3 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.1) (0.9.4)
Requirement already satisfied, skipping upgrade: Flask-Principal>=0.3.3 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.1) (0.4.0)
Requirement already satisfied, skipping upgrade: passlib>=1.7 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.1) (1.7.2)
Requirement already satisfied, skipping upgrade: Flask-WTF>=0.13.1 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.1) (0.14.3)
Requirement already satisfied, skipping upgrade: Flask-Mail>=0.7.3 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Security->genotype==4.1.1) (0.9.1)
Requirement already satisfied, skipping upgrade: pydash>=3.4.7 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Alchy->genotype==4.1.1) (4.8.0)
Requirement already satisfied, skipping upgrade: six in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.1) (1.15.0)
Requirement already satisfied, skipping upgrade: oauthlib in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.1) (3.1.0)
Requirement already satisfied, skipping upgrade: requests>=2.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.1) (2.24.0)
Requirement already satisfied, skipping upgrade: urlobject in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.1) (2.4.3)
Requirement already satisfied, skipping upgrade: requests-oauthlib>=1.0.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Dance==3.0.0->genotype==4.1.1) (1.3.0)
Requirement already satisfied, skipping upgrade: humanfriendly>=7.1 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from coloredlogs->genotype==4.1.1) (8.2)
Requirement already satisfied, skipping upgrade: visitor in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Bootstrap->genotype==4.1.1) (0.1.3)
Requirement already satisfied, skipping upgrade: dominate in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Bootstrap->genotype==4.1.1) (2.5.2)
Requirement already satisfied, skipping upgrade: speaklater>=1.2 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.1) (1.3)
Requirement already satisfied, skipping upgrade: Babel>=1.0 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.1) (2.8.0)
Requirement already satisfied, skipping upgrade: blinker in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-Principal>=0.3.3->Flask-Security->genotype==4.1.1) (1.4)
Requirement already satisfied, skipping upgrade: WTForms in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Flask-WTF>=0.13.1->Flask-Security->genotype==4.1.1) (2.3.3)
Requirement already satisfied, skipping upgrade: chardet<4,>=3.0.2 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.1) (3.0.4)
Requirement already satisfied, skipping upgrade: certifi>=2017.4.17 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.1) (2020.6.20)
Requirement already satisfied, skipping upgrade: idna<3,>=2.5 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.1) (2.10)
Requirement already satisfied, skipping upgrade: urllib3!=1.25.0,!=1.25.1,<1.26,>=1.21.1 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from requests>=2.0->Flask-Dance==3.0.0->genotype==4.1.1) (1.25.10)
Requirement already satisfied, skipping upgrade: pytz>=2015.7 in /home/proj/bin/conda/envs/P_genotype/lib/python3.7/site-packages (from Babel>=1.0->Flask-BabelEx>=0.9.3->Flask-Security->genotype==4.1.1) (2020.1)
Building wheels for collected packages: genotype
  Building wheel for genotype (PEP 517) ... done
  Created wheel for genotype: filename=genotype-4.1.1-py3-none-any.whl size=31298 sha256=355b92d7302841cfc4b014a2075aef6ab8fe5f28fc483d2a1126ff8064ef63ee
  Stored in directory: /home/henrik.stranneheim/tmp/pip-ephem-wheel-cache-zyc_hqle/wheels/63/d7/38/2d616f160ea626015965a94d5e26e4288d169a24a3f5151da1
Successfully built genotype
Installing collected packages: genotype
  Attempting uninstall: genotype
    Found existing installation: genotype 4.1.1
    Uninstalling genotype-4.1.1:
      Successfully uninstalled genotype-4.1.1
Successfully installed genotype-4.1.1
fatal: Not a git repository (or any parent up to mount point /home)
Stopping at filesystem boundary (GIT_DISCOVERY_ACROSS_FILESYSTEM not set).
repository is clean
Logging deploy ...
Getting deployer... done.
Getting last commit message and SHA... done.
Getting version of deploy scripts... /home/henrik.stranneheim
done.
Log deploy... done.
No broken requirements found.
genotype, version 4.1.1
```
